cpp/src/prims/detail/nbr_intersection.cuh

```diff
@@ -1003,7 +1234,37 @@ nbr_intersection(raft::handle_t const& handle,
         rx_v_pair_nbr_intersection_indices.resize(
           rx_v_pair_nbr_intersection_offsets.back_element(handle.get_stream()),
           handle.get_stream());
+
+        optional_property_buffer_t rx_v_pair_optional_nbr_intersection_properties0{};
+        optional_property_buffer_t rx_v_pair_optional_nbr_intersection_properties1{};
+        optional_idx_buffer_t rx_v_pair_optional_nbr_intersection_idx_buffer{};
```

Comment:
These are view objects, right? (not owning buffers, `..._buffer_t` might be confusing.